doc/getting_started/toolchain_custom_cmake.rst

```diff
@@ -39,5 +43,21 @@ your :makevar:`ZEPHYR_TOOLCHAIN_VARIANT`, :makevar:`TOOLCHAIN_ROOT`, and other
 settings in a file named :file:`my-toolchain.cmake`, you can then invoke cmake
 as ``cmake -C my-toolchain.cmake ...`` to save typing.
 
+In order for the custom toolchain to include a toolchain specific header file,
+the setting of :envvar:`TOOLCHAIN_ROOT` will define ``__TOOLCHAIN_CUSTOM__``
+which will force :file:`include/toolchain.h` to include the
+:file:`toolchain/other.h`. The :file:`other.h` must be available
```

Comment:
If following the other suggestion, then this would become something like:
> Zephyr includes `includes/toolchain.h` which again includes a toolchain specific header based on the compiler identifier, such as `__llvm__` or `__GNUC__`.
..**note**::
Some custom compilers identifies themselves as the compiler on which they are based, for example __llvm__, and thus includes the file `toolchain/llvm.h`. To use a custom `toolchain/other.h` header and disable the default Zephyr toolchain header then add the line: `set(TOOLCHAIN_USE_CUSTOM 1)` in the file `cmake/toolchain/<toolchain name>/generic.cmake`.

Reply:
Documentation rewritten to slightly different phrasing and wording, but hopefully it is acceptable.